data_models.py

```diff
@@ -152,12 +152,11 @@ class MainConfig:
     mirrors_dir: str
     vault_mirror: str
     versions: list[str] = field(default_factory=list)
-    optional_module_versions: dict[list] = field(default_factory=dict)
-    arches: list = field(default_factory=list)
+    optional_module_versions: dict[str, list[str]] = field(
+        default_factory=dict
+    )
+    arches: dict[str, list] = field(default_factory=dict)
     duplicated_versions: dict[str, str] = field(default_factory=dict)
     vault_versions: list[str] = field(default_factory=list)
-    versions_arches: dict[str, list[str]] = field(
```

Comment:
Per our chat, this is fine for the 10-beta but it removes the ability to pin repo support between minor versions and architectures for optional modules.  This will likely be needed for EPEL10 on v2 if we choose to rebuild it for v2 so we'll have to revisit this in the near future.